aea/aea_builder.py

```diff
@@ -0,0 +1,746 @@
+            return
+
+        if component_id.component_type == ComponentType.PROTOCOL:
+            self._resources.remove_protocol(component_id.public_id)
+        elif component_id.component_type == ComponentType.SKILL:
+            self._resources.remove_skill(component_id.public_id)
+
+    def remove_component(self, component_id: ComponentId) -> "AEABuilder":
+        """Remove a component."""
+        self._check_can_remove(component_id)
+        self._remove(component_id)
+        return self
+
+    def _remove(self, component_id: ComponentId):
+        self._package_dependency_manager.remove_component(component_id)
+        self._remove_component_from_resources(component_id)
+
+    def add_protocol(self, directory: PathLike) -> "AEABuilder":
+        """Add a protocol to the agent."""
+        self.add_component(ComponentType.PROTOCOL, directory)
+        return self
+
+    def remove_protocol(self, public_id: PublicId) -> "AEABuilder":
+        """Remove protocol"""
+        self.remove_component(ComponentId(ComponentType.PROTOCOL, public_id))
+        return self
+
+    def add_connection(self, directory: PathLike) -> "AEABuilder":
+        """Add a protocol to the agent."""
+        self.add_component(ComponentType.CONNECTION, directory)
+        return self
+
+    def remove_connection(self, public_id: PublicId) -> "AEABuilder":
+        """Remove a connection"""
+        self.remove_component(ComponentId(ComponentType.CONNECTION, public_id))
+        return self
+
+    def add_skill(self, directory: PathLike) -> "AEABuilder":
+        """Add a skill to the agent."""
+        self.add_component(ComponentType.SKILL, directory)
+        return self
+
+    def remove_skill(self, public_id: PublicId) -> "AEABuilder":
+        """Remove protocol"""
+        self.remove_component(ComponentId(ComponentType.SKILL, public_id))
+        return self
+
+    def add_contract(self, directory: PathLike) -> "AEABuilder":
+        """Add a contract to the agent."""
+        self.add_component(ComponentType.CONTRACT, directory)
+        return self
+
+    def remove_contract(self, public_id: PublicId) -> "AEABuilder":
+        """Remove protocol"""
+        self.remove_component(ComponentId(ComponentType.CONTRACT, public_id))
+        return self
+
+    def _build_identity_from_wallet(self, wallet: Wallet) -> Identity:
+        """Get the identity associated to a wallet."""
+        assert self._name is not None, "You must set the name of the agent."
+        if len(wallet.addresses) > 1:
+            identity = Identity(
+                self._name,
+                addresses=wallet.addresses,
+                default_address_key=self._default_ledger,
+            )
+        else:  # pragma: no cover
+            identity = Identity(
+                self._name, address=wallet.addresses[self._default_ledger],
+            )
+        return identity
+
+    def _process_connection_ids(
```

Comment:
this emulates the option `--connections` in `aea run`

Reply:
it is called in build. 
It makes me think that maybe the `connection_ids` argument should be handled by `Agent.start()` instead.